File: spoj/classicals/BDOI16B.py
inf = float("inf")

size=10**6 + 1
isprime = [1]*size
primes = [2]
isprime[0] = isprime[1] = 0
for i in xrange(4,size,2):
	isprime[i]=0
for i in xrange(3,int(size**0.5)+1,2):
	if isprime[i]:
		for j in xrange(i*i,size,2*i):
			isprime[j] = 0
for i in xrange(3,size,2):
	if isprime[i]: primes.append(i)

def solve(n,k):
	tk = k
	ans = 0
	while tk <= n:
		ans += n/tk
		tk *= k
	return ans
case = 0
for _ in xrange(int(raw_input())):
	

File: spoj/classicals/archive/BDOI16B.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long ll ;
typedef map<ll ,ll >mll ;
typedef vector<int>vi;
const ll inf=LONG_MAX;
const ll SIZE=5000001;
ll isprime[SIZE];
vi primes;
void sieve(){
isprime[0]=isprime[1]=1;
for(ll i=4;i<SIZE;i+=2){
isprime[i]=1;
}
for(ll i=3;i<sqrt(SIZE)+1.5;i+=2){
if(!isprime[i]){
for(ll j=i*i;j<SIZE;j+=2*i){
isprime[j]=1;
}
}
}
primes.push_back(2);
for(ll i=3;i<SIZE;i+=2){
if(!isprime[i])primes.push_back(i);
}
}
ll solve(ll n,ll k){
ll tk=k;ll ans=0;
while(tk<=n){
ans+=n/tk;
tk*=k;
}
return ans;
}
int main(void){
ios::sync_with_stdio(false);
sieve();
ll t;
cin>>t;
ll C=1;
while(t--){
ll n,k;
cin>>n>>k;
mll fact;
ll tk=k;
for(ll prime:primes){
if(tk<2)break;
while(tk%prime==0){
fact[(ll )prime]++;
tk/=prime;
}
}
ll ans=inf;
for(auto it=fact.begin();it!=fact.end();it++){
ll factor=it->first;
ans=min(ans,solve(n,factor)/fact[factor]);
}
cout<<ans<<endl;
}

}

